### PR TITLE
Change device output path on Android

### PR DESCRIPTION
**Pull request by oxve**
b/289417243

### CODE REVIEW

File: starboard/android/shared/launcher.py [hunk 1]
```diff
@@ -450,4 +450,4 @@ def GetDeviceIp(self):
 
   def GetDeviceOutputPath(self):
     """Writable path where test targets can output files"""
-    return f'/data/data/{_APP_PACKAGE_NAME}/cache/'
+    return f'/data/data/{_APP_PACKAGE_NAME}/log/'
```
Comment (kaidokert) [inline starboard/android/shared/launcher.py:453]: This currently controls only where the gtests Junit files get put, correct ? If so, lgtm

**Review by kaidokert — APPROVED** [review]
<typo>

File: starboard/android/shared/launcher.py (hunk 1 above)
Comment (oxve) [reply starboard/android/shared/launcher.py:453]: I don't see any other uses for this function. I need to coordinate this with a change in Mobile Harness to merge to ensure that it knows where to look for the files.

Comment (kaidokert) [reply starboard/android/shared/launcher.py:453]: Ack - keep in mind "old branches" as you go through this. Although tbh we mostly care if it's correct on trunk+24 lts at this point

Comment (oxve) [reply starboard/android/shared/launcher.py:453]: I think this change should be safe to cherry pick to release branches.

I'm unresolving this conversation so this doesn't get accidentally merged before the change to MH has been lined up.

Comment (y4vor) [reply starboard/android/shared/launcher.py:453]: Isn't the actual path "cache/log" ?
https://github.com/youtube/cobalt/blob/main/starboard/android/shared/system_get_path.cc#L118

**Pull request closed without merging**